scripts/lib/CIME/SystemTests/system_tests_common.py

```diff
@@ -35,6 +35,9 @@ def __init__(self, case, expected=None):
         self._init_environment(caseroot)
         self._init_locked_files(caseroot, expected)
 
+    def __del__(self):
+        self._case.set_value("RUN_WITH_SUBMIT", False)
```

Comment:
Is this necessary? I added some code around line 250 in case_run to handle RUN_WITH_SUBMIT.